### PR TITLE
#1984 Fix Holder Binding Validation Handling in vc-verifier in develop

### PR DESCRIPTION
**Pull request by mayuradesh**
## Summary by CodeRabbit

* **Documentation**
  * Updated release notes for import guidance (1.5.x onwards) and added 1.8.x+ docs covering ECC R1/K1 and holder-binding behavior; improved Maven integration heading.

* **New Features**
  * Added ECC R1/K1 support.
  * Expanded holder-binding validation to support did:key and did:jwk holder proofs.

* **Bug Fixes**
  * More resilient handling of presentations missing verifiable credentials.
  * Clearer verification failure reporting for holder-binding issues.

* **Tests**
  * Expanded test coverage for holder-binding scenarios and presentation edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/inji/vc-verifier/pull/254?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,8 +88,99 @@ related to validating and verifying Verifiable Credentials (VCs). It also perfor
...
+### From Version `release-1.8.x` onwards:
+---
+
+1. Added **ECC R1 and K1 support** in both `CredentialVerifier.kt` & `PresentationVerifier.kt`.
+2. Added **Holder Binding Check Support**  in `PresentationVerifier.kt`.
+
+### Holder Binding Validation
+---
+From Version `release-1.8.1` onwards, `PresentationVerifier.kt` includes improved holder binding validation handling for `did:key` and `did:jwk` DID methods in existing V1 and V2 verification APIs.
+
```
[the remark this answers is not in the document]
Comment (mayuradesh) [reply README.md:102]: It says release-1.8.1 onwards so + is implied

File: vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/PresentationVerifier.kt [hunk 2]
```diff
@@ -171,103 +175,242 @@ class PresentationVerifier {
...
+
+                subjects.forEach subjectLoop@ { subject ->
+
+                    val subjectStr =
+                        (subject as? Map<*, *>)?.get(ID) as? String
+
+                    if (subjectStr == null) {
+                        logger.info("Skipping subject binding check: subject.id missing")
+                        return@subjectLoop
+                    }
```
[the remark this answers is not in the document]
Comment (mayuradesh) [reply vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/PresentationVerifier.kt:255]: @coderabbitai this is to support credential which do not have subject.id, also ReadMe explains this

**Comment by mayuradesh** [comment]
@coderabbitai review this again

**Comment by mayuradesh** [comment]
@coderabbitai any comments

File: vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/PresentationVerifier.kt [hunk 3]
```diff
@@ -78,6 +80,7 @@ class PresentationVerifier {
         val presentationVerificationStatus: VPVerificationStatus = getPresentationVerificationStatus(presentation)
 
         val verifiableCredentials = JSONObject(presentation).getJSONArray(KEY_VERIFIABLE_CREDENTIAL)
+
```
[the remark this answers is not in the document]
Comment (mayuradesh) [reply vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/PresentationVerifier.kt:83]: validateHolderBindingForDidKeyAndJwk() is correctly handling both missing and empty KEY_VERIFIABLE_CREDENTIAL

**Review by Prafulrakhade — APPROVED** [review]
